components/plan-build-ps1/bin/hab-plan-build.ps1

```diff
@@ -1211,7 +1211,19 @@ function Invoke-DefaultBuildConfig {
     }
     if (Test-Path "$PLAN_CONTEXT/hooks") {
         Write-BuildLine "Writing hooks"
-        Copy-Item "$PLAN_CONTEXT/hooks" $pkg_prefix -Recurse
+        # If a .ps1 hook is present, copy that otherwise copy an
+        # extensionless hook and ignore other extensions
+        # The supervisor does not recognize extensions so all hooks are
+        # copied without extensions
+        $hooksDir = Join-Path $pkg_prefix hooks
+        New-Item $hooksDir -ItemType Directory
+        foreach ($file in Get-ChildItem "$PLAN_CONTEXT/hooks") {
+            if($file.Extension -eq ".ps1") {
+                Copy-Item $file (Join-Path $hooksDir $file.BaseName)
+            } elseif (!(Test-Path (Join-Path $hooksDir $file.BaseName)) -and ($file.Extension -eq "")) {
+                Copy-Item $file (Join-Path $hooksDir $file.BaseName)
+            }
```

Comment:
These could be combined in a single `if` statement with an `or`, but maybe you intentionally separated them in order to make the logic easier to follow?